swan-lake/resources/featured-scenarios/build-a-change-data-capture-service-in-ballerina.md

```diff
@@ -0,0 +1,299 @@
+import ballerinax/mysql.cdc.driver as _;
+
+configurable string username = ?;
+configurable string password = ?;
+
+listener mysql:CdcListener financeDBListener = new (
+    database = {
+        username,
+        password,
+        includedDatabases: "finance_db",
+        includedTables: "finance_db.transactions"
+    },
+    options = {
+        snapshotMode: cdc:NO_DATA,
+        skippedOperations: [cdc:TRUNCATE, cdc:UPDATE, cdc:DELETE]
+    }
+);
+```
+
+> **Tip:** Use [Ballerina configurable variables](https://ballerina.io/learn/configure-a-sample-ballerina-service/) to manage credentials securely.
+
+*Define the CDC Service*
+
+```ballerina
+service cdc:Service on financeDBListener {
+    isolated remote function onCreate(Transactions trx) returns error? {
+        log:printInfo(`Transaction with id: ${trx.tx_id}`);
+        if trx.amount > 10000.0 {
+            log:printWarn(
+                `⚠️ Fraud alert: High-value transaction detected: 
+                Id: ${trx.tx_id}, 
+                User Id: ${trx.user_id}, 
+                Amount: $${trx.amount}`
+            );
+        }
+    }
+
+    isolated remote function onError(cdc:Error e) {
+        log:printError("CDC error occurred", e);
+    }
+}
+```
+
+> **Note:** We're only handling `onCreate` here, but the service can also support `onRead`, `onUpdate`, and `onDelete`.
+
+## The complete code
+
+Here’s the complete `main.bal`:
+
+```ballerina
+import ballerina/log;
+import ballerinax/cdc;
+import ballerinax/mysql;
+import ballerinax/mysql.cdc.driver as _;
+
+configurable string username = ?;
+configurable string password = ?;
+
+type Transactions record {|
+    int tx_id;
+    int user_id;
+    float amount;
+    string status;
+    int created_at;
+|};
+
+listener mysql:CdcListener financeDBListener = new (
+    database = {
+        username,
+        password,
+        includedDatabases: "finance_db",
+        includedTables: "finance_db.transactions"
+    },
+    options = {
+        snapshotMode: cdc:NO_DATA,
+        skippedOperations: [cdc:TRUNCATE, cdc:UPDATE, cdc:DELETE]
+    }
+);
+
+service cdc:Service on financeDBListener {
+    isolated remote function onCreate(Transactions trx) returns error? {
+        log:printInfo(`Transaction with id: ${trx.tx_id}`);
+        if trx.amount > 10000.0 {
+            log:printWarn(
+                `⚠️ Fraud alert: High-value transaction detected: 
+                Id: ${trx.tx_id}, 
+                User Id: ${trx.user_id}, 
+                Amount: $${trx.amount}`
+            );
+        }
+    }
+
+    isolated remote function onError(cdc:Error e) {
+        log:printError("CDC error occurred", e);
+    }
+}
+```
+
+## Run the service
+
+### Add the credentials
+
+In the package directory, create a new file named `Config.toml` and specify the configurations below to connect to the MySQL database.
+
+```toml
+username="root"
+password="rootPassword"
+```
+
+### Execute the `bal run` command
+
+```bash
+bal run
```

Comment:
```suggestion
$ bal run
```

Reply:
Fixed in [098a221](https://github.com/ballerina-platform/ballerina-dev-website/pull/9891/commits/098a221ea98fef64881aca9b621b8c3733ff0ff3)